backend/tests/v5/e2e/routes/teamspaces/projects/containers/containers.test.js

```diff
@@ -276,6 +304,99 @@ const testDeleteFavourites = () => {
+				.send({ name: 'name' }).expect(templates.notAuthorized.status);
+			expect(res.body.code).toEqual(templates.notAuthorized.code);
+		});
+
+		test('should fail if the user does not have adequate permissions to edit the container (viewer)', async () => {
+			const res = await agent.patch(`${route(modelWithViews._id)}?key=${users.viewer.apiKey}`)
+				.send({ name: 'name' }).expect(templates.notAuthorized.status);
+			expect(res.body.code).toEqual(templates.notAuthorized.code);
+		});
+
+		test('should fail if the user does not have adequate permissions to edit the container (commenter)', async () => {
+			const res = await agent.patch(`${route(modelWithViews._id)}?key=${users.commenter.apiKey}`)
+				.send({ name: 'name' }).expect(templates.notAuthorized.status);
+			expect(res.body.code).toEqual(templates.notAuthorized.code);
+		});
+
+		test('should fail if the project does not exist', async () => {
+			const res = await agent.patch(`/v5/teamspaces/${teamspace}/projects/dflkdsjfs/containers/${modelWithViews._id}?key=${users.tsAdmin.apiKey}`)
+				.send({ name: 'name' }).expect(templates.projectNotFound.status);
+			expect(res.body.code).toEqual(templates.projectNotFound.code);
+		});
+
+		test('should fail if the container does not exist', async () => {
+			const res = await agent.patch(`/v5/teamspaces/${teamspace}/projects/${project.id}/containers/dfsfaewfc?key=${users.tsAdmin.apiKey}`)
+				.send({ name: 'name' }).expect(templates.containerNotFound.status);
+			expect(res.body.code).toEqual(templates.containerNotFound.code);
+		});
+
+		test('should fail if a body param is not of the expected type', async () => {
+			const res = await agent.patch(`${route(modelWithViews._id)}?key=${users.tsAdmin.apiKey}`)
+				.send({ name: 123 }).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the body of the request contains extra payload', async () => {
+			const res = await agent.patch(`${route(modelWithViews._id)}?key=${users.tsAdmin.apiKey}`)
+				.send({ name: 'name', extra: 123 }).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the defaultView is not found', async () => {
+			const res = await agent.patch(`${route(modelWithViews._id)}?key=${users.tsAdmin.apiKey}`)
+				.send({ name: 'name', defaultView: '374bb150-065f-11ec-8edf-ab0f7cc84da8' }).expect(templates.viewNotFound.status);
+			expect(res.body.code).toEqual(templates.viewNotFound.code);
+		});
+
+		test('should fail if the defaultLegend is not found', async () => {
+			const res = await agent.patch(`${route(modelWithViews._id)}?key=${users.tsAdmin.apiKey}`)
+				.send({ name: 'name', defaultLegend: '374bb150-065f-11ec-8edf-ab0f7cc84da8' }).expect(templates.legendNotFound.status);
+			expect(res.body.code).toEqual(templates.legendNotFound.code);
+		});
+
```

Comment:
add Test 

- should update settings if not all of the fields are provided (just one test is fine, don't need to test all combinations!)